utils/db_helper.js

```diff
@@ -172,6 +174,20 @@ async function rollbackDBTransaction(client) {
   client.release();
 }
 
+async function runInTransaction(commands) {
+  const client = await createDBTransaction();
+  let results = [];
+  try {
+    for (const command of commands) {
```

Comment:
I like it! should we move all function that require transaction to use this?

Reply:
yes that would remove a lot of duplicate transaction creating and error handling code